### PR TITLE
feat: 231 add block stream tools

### LINKED ISSUE #231: Add block stream tools
### Problem

We need some utility tools for processing and performance testing block streams

### Solution

Add new subproject with tools

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by jasperpotts**
**Description**:
Adds new sub project for command line block stream tools :-) 

**Related issue(s)**:

Fixes #231 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by mattp-swirldslabs — APPROVED** [review]
Do you mind adding a README.md with some of the common usages?

**Review by ata-nas — COMMENTED** [review]
I have a couple of problems with building, the CI also seems to fail with the same, otherwise looks good, I tested both commands, seem to do what they should be doing.

File: tools/src/main/java/com/hedera/block/tools/commands/BlockInfo.java [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+     * @return the info string
+     */
+    public String blockInfo(Path blockProtoFile) {
+        try (InputStream fIn = Files.newInputStream(blockProtoFile)) {
+            byte[] uncompressedData;
+            if (blockProtoFile.getFileName().toString().endsWith(".gz")) {
+                uncompressedData = new GZIPInputStream(fIn).readAllBytes();
+            } else {
+                uncompressedData = fIn.readAllBytes();
+            }
```
Comment (ata-nas) [inline tools/src/main/java/com/hedera/block/tools/commands/BlockInfo.java:209]: This branch is currently quite behind, after rebasing you will have access to the `common` module which already has a method that will read file bytes unsafe: `com.hedera.block.common.utils.FileUtilities#readFileBytesUnsafe(Path)` I suggest that you use that.

File: tools/src/main/java/com/hedera/block/tools/commands/ConvertToJson.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        final String fileName = blockProtoFile.getFileName().toString();
+        final String fileNameNoExt = fileName.substring(0, fileName.lastIndexOf('.'));
+        final Path outputFile = blockProtoFile.resolveSibling(fileNameNoExt + ".json");
+        try (InputStream fIn = Files.newInputStream(blockProtoFile)) {
+            byte[] uncompressedData;
+            if (blockProtoFile.getFileName().toString().endsWith(".gz")) {
+                uncompressedData = new GZIPInputStream(fIn).readAllBytes();
+            } else {
+                uncompressedData = fIn.readAllBytes();
+            }
```
Comment (ata-nas) [inline tools/src/main/java/com/hedera/block/tools/commands/ConvertToJson.java:91]: Same as in `BlockInfo`:

This branch is currently quite behind, after rebasing you will have access to the `common` module which already has a method that will read file bytes unsafe: `com.hedera.block.common.utils.FileUtilities#readFileBytesUnsafe(Path)` I suggest that you use that.

File: tools/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+    id("application")
+    id("com.hedera.block.tools")
+//    id("com.github.johnrengelman.shadow") version "8.1.1"
```
[the remark this answers is not in the document]
Comment (jsync-swirlds) [reply tools/build.gradle.kts:20]: Remove in this case; we shouldn't be using this dependency.

File: tools/src/main/java/com/hedera/block/tools/commands/ConvertToJson.java (hunk 2 above)
Comment (jasperpotts) [reply tools/src/main/java/com/hedera/block/tools/commands/ConvertToJson.java:99]: maybe later

**Review by ata-nas — APPROVED** [review]
From my side, my comments are resolved. The PR Gradle Check fails due to spotlessApply, maybe you could run that in another commit so it can pass?

File: .github/workflows/release-automation.yaml [hunk 4]
```diff
@@ -44,7 +47,7 @@ jobs:
 
     steps:
       - name: Harden Runner
-        uses: step-security/harden-runner@91182cccc01eb5e619899d80e4e971d6181294a7 # v2.10.1
```
Comment (mishomihov00) [inline .github/workflows/release-automation.yaml:47]: Why downgrading the version?

File: .github/workflows/release-automation.yaml [hunk 5]
```diff
@@ -77,18 +80,17 @@ jobs:
         with:
           fetch-depth: 0
           ref: main
-          token: ${{ secrets.GH_ACCESS_TOKEN }}
-          persist-credentials: 'true'
+          token: ${{ secrets.HEDERA_BOT_TOKEN }}
 
       - name: Import GPG Key
         id: gpg_importer
-        uses: step-security/ghaction-import-gpg@6c8fe4d0126a59d57c21f87c9ae5dd3451fa3cca # v6.1.0
```
Comment (mishomihov00) [inline .github/workflows/release-automation.yaml:85]: Use the step-security maintained version of this action.

File: .github/workflows/release-automation.yaml [hunk 6]
```diff
@@ -177,19 +178,17 @@ jobs:
       - name: Checkout Repository
         uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
         with:
-          fetch-depth: 0
           ref: main
-          token: ${{ secrets.GH_ACCESS_TOKEN }}
 
       - name: Import GPG Key
         id: gpg_importer
-        uses: step-security/ghaction-import-gpg@6c8fe4d0126a59d57c21f87c9ae5dd3451fa3cca # v6.1.0
```
Comment (mishomihov00) [inline .github/workflows/release-automation.yaml:186]: Use the step-security maintained version of this action.

**Review by mishomihov00 — CHANGES_REQUESTED** [review]
My review applies only to
`.github/workflows/release-automation.yaml`

File: buildSrc/src/main/kotlin/com.hedera.block.tools.gradle.kts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+plugins {
+    id("application")
+    id("com.hedera.block.conventions")
+    id("me.champeau.jmh")
+} //
+// val maven = publishing.publications.create<MavenPublication>("maven") { from(components["java"])
+// }
+//
+// signing.sign(maven)
```
Comment (mattp-swirldslabs) [inline buildSrc/src/main/kotlin/com.hedera.block.tools.gradle.kts:25]: Can we remove these comments? If they're needed for an operation, can you document that?

File: tools/README.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+**Options:**
+- `-c` or `--csv`
+  - Enable CSV output mode (default: false)
+- `-ms <minSizeMb>` or `--min-size=<minSizeMb>`
+  - Filter to only files bigger than this minimum file size in megabytes
+- `-o <outputFile>` or `--output-file=<outputFile>`
+  - Output to file rather than stdout
+- `<files>...`
+  - The block files or directories of block files to print info for
```
Comment (mattp-swirldslabs) [inline tools/README.md:45]: Thanks for the README.md!

**Review by mattp-swirldslabs — APPROVED** [review]
Nice work, thanks for the `README.md`. I just had a question about some commented out code.

**Review by mattp-swirldslabs — APPROVED** [review]
Looks good

**Review by jsync-swirlds — APPROVED** [review]
Looks Good, just one minor question that can be addressed in a later PR, if desired.

**Pull request merged**